src/main/scala/edu/berkeley/ce/rockslicing/RockSlicer.scala

```diff
@@ -60,15 +64,35 @@ object RockSlicer {
       !block.faces.exists(_.processorJoint)
     }
 
-    // Process processor blocks before continueing with remaining blocks
-    // Collect all blocks that contain processor joints from all nodes
-    val allProcessorBlocks = processorBlocks.collect()
+    // Process processor blocks before continuing with remaining blocks
     // Search blocks for matching processor joints
-    if (allProcessorBlocks.length > 0) {
-      val reconstructedBlocks = mergeBlocks(allProcessorBlocks, Seq.empty[Block], globalOrigin)
+    if (processorBlocks.count() > 0) {
+      // Create pair RDD that contains initially all the blocks on each partion as the keys and
+      // an empty sequence of blocks as the values. The keys represent blocks that are yet to be merged,
```

Comment:
I can change this. Thought using the "key" and "value" descriptions seemed appropriate based on pair RDD's, but I can see how that is confusing since it's nothing like the traditional case.